packages/patternfly-4/react-core/src/components/Select/SelectOption.tsx

```diff
@@ -0,0 +1,154 @@
+import * as React from 'react';
+import styles from '@patternfly/react-styles/css/components/Select/select';
+import { default as checkStyles } from '@patternfly/react-styles/css/components/Check/check';
+import { css } from '@patternfly/react-styles';
+import { CheckIcon } from '@patternfly/react-icons';
+import { SelectConsumer, SelectVariant, KeyTypes } from './selectConstants';
+
+export interface SelectOptionProps {
+  /** Additional classes added to the Select Option */
+  className?: string;
+  /** Internal index of the option */
+  index?: number;
+  /** The value for the option */
+  value: string;
+  /** Flag indicating if the option is disabled */
+  isDisabled?: boolean;
+  /** Flag indicating if the option acts as a placeholder */
+  isPlaceholder?: boolean;
+  /** Internal flag indicating if the option is selected */
```

Comment:
What does `Internal` flag mean?

Reply:
It means the property is used by the other select classes, and isn't one that has to be passed in by the user. Our docs generate all props so I wanted to specify which aren't expected.

Reply:
I opened #2380 for this.